_projects/i_gitt_scm_support.md

```diff
@@ -0,0 +1,58 @@
+---
+name: "IGitt SCM support"
+desc: "Add support for git hosters in IGitt and convert coala VCS bears to
+       use IGitt"
+requirements:
+  - "An enhancement to the coala VCS bears"
+  - "An enhancement to IGitt"
+initiatives:
+  - GSoC
+collaborating_projects:
+  - "coala"
+  - "IGitt"
+---
+
+[IGitt](https://gitlab.com/gitmate/open-source/IGitt) is a simple library
+that allows you to access various git hosting services like GitHub, GitLab
+and so on via one unified python interface.
+
+The coala bears project provides support for git and mercurial VCS but it
+does not use IGitt. With this project, VCS in coala-bears should be
+simplified using IGitt.
+
+However, there are some git hosters that are not yet supported by IGitt.
+One such hoster is BitBucket. So it is necessary to add BitBucket support to
+IGitt before using it to convert the VCS bears in coala-bears.
+
+Other git hosters like Gerrit and Phabricator are important for IGitt adoption 
+into large OSS organizations. So are some other non-git SCM, such as Fossil
+and Launchpad. 
+
+Support for all these git hosters should be added to IGitt.
+
+A recommended preparation task in addition to already mentioned requirements
+for this project is to make an enhancement to https://github.com/guyzmo/git-repo
+
+#### Milestones
+
+##### GSOC 2018 COMMUNITY/BONDING
+
+* Create a cEP describing the details of how support for additional git
+  hosters will be added to IGitt.
+* Understand how bears for VCS are implemented in coala bears and describe
+  details of how IGitt will be used to improve them.
+* The participant has a merged enhancement to IGitt.
+* The participant has completed the newcomer process.
+
+##### CODING PHASE 1
+
+* Add BitBucket (git & mecurial) support to IGitt.
```

Comment:
Same thing applies, have a look at the other comment.

Reply:
And here also.

Reply:
@kriti21 , it would be helpful to indicate what features of IGitt would be useful for the VCS bears .

One helpful enhancement is https://gitlab.com/gitmate/open-source/IGitt/issues/111 , or similar, to convert a VCS `remote` string (e.g. `git@github.com/...`) to an object which includes identifying the hoster from the domain name in the VCS.